openresty-build-tools/kong-ngx-build

```diff
@@ -600,41 +618,89 @@ main() {
     fi
   fi
 
-  if [ -n "$BORINGSSL_VER" ]; then
+  if [ "$BORINGSSL_VER" != 0 -a "$SSL_PROVIDER" = "boringssl" ]; then
     if [ ! -d "$BORINGSSL_DOWNLOAD/.openssl/lib" ]; then
       notice "Building BoringSSL..."
 
+      arch=$(uname -m)
+      if [ "$(arch)" != "x86_64" ]; then
+        echo "Error: BoringSSL builds only supported in x86_64; got $arch" >&2
+        exit 1
+      fi
```

Comment:
dissapointed but acceptable for now

Reply:
I am unsure the boringssl integrity checks work on arm64 boxes. On my m1, it did not. We will need to circle back.